app/models/exchange.rb

```diff
@@ -118,4 +120,11 @@ def clone_all_exchange_variants(exchange_id)
       variant_ids.map{ |variant_id| { variant_id: variant_id, exchange_id: exchange_id } }
     )
   end
+
+  def delete_related_exchange_variants
+    ExchangeVariant.where(variant_id: variant_ids).
+      joins(:exchange).
+      where(exchanges: { order_cycle: order_cycle, incoming: false }).
+      delete_all
+  end
```

Comment:
I think that the same applies here. We should only delete outgoing variants if this is an incoming exchange.